src/js/components/search/filters/awardType/PrimaryAwardType.jsx

```diff
@@ -23,29 +22,31 @@ export default class PrimaryAwardType extends React.Component {
     }
 
     toggleSubItems() {
-        if (this.state.showSubItems === false) {
-            this.setState({ showSubItems: true });
-        }
-        else {
-            this.setState({ showSubItems: false });
-        }
+        this.setState({
+            showSubItems: !this.state.showSubItems
+        });
     }
 
     render() {
-        const primaryAwardList = (<div className="primaryAwardTypeOption">
-            <input type="checkbox" id={this.props.name} value={this.props.name} />
-            <label htmlFor={this.props.name}>{this.props.name}</label>
-            <a className="toggle" href="#null" onClick={this.toggleSubItems.bind(this)}>
-                <Icons.AngleDown />
-            </a>
-            {this.state.showSubItems === true ?
-                <SecondaryAwardType subList={this.props.subList} /> : null }
-        </div>);
+        let primaryAward = (<CollapsedAwardType
+            name={this.props.name}
+            click={this.toggleSubItems.bind(this)} />);
+
+        const secondaryAwardTypes = this.props.subList.map((subList, index) =>
```

Comment:
Your `render` function needs to be as light as possible because of how frequently it may be executed. In the current logic, you are always looping through the secondary award types, regardless of whether or not you need to (ie, the filter is collapsed). But if you move the loop inside of the `if` block, it only runs on demand.